common/src/main/java/io/crate/types/TypeSignature.java

```diff
@@ -160,7 +166,7 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
```

Comment:
why is this removed?

Reply:
I was about to ask a question here. Did it to be able to compare the parameters with NamedTypeSignature and TypeSignature, which should probably never be compared.

eg. parseSignature(object(text int)) is(ObjectType.getSignature) 

will adjust tests.